### PR TITLE
Missing ssrMode setting in ssr Quick Example / Fix Typescript type

### PR DESCRIPTION
**Pull request by helloguille**
Without ssrMode enabled the system will not dispatch the right Promise on cache-miss.

### What does this PR do?

Updates documentation for ssr.

### Checklist

- [Y] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [Y] I have added or updated any relevant documentation
- [N] I have added or updated any relevant tests

### CODE REVIEW

**Comment by helloguille** [comment]
Added a new Typescript type fix.

File: packages/graphql-hooks-ssr/README.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ app.get('/', async (req, reply) => {
   const client = new GraphQLClient({
     url: 'https://domain.com/graphql',
     cache: memCache() // NOTE: a cache is required for SSR,
-    fetch
+    fetch,
+    ssrMode: true
```
Comment (bmullan91) [inline packages/graphql-hooks-ssr/README.md]: This isn't required when using `graphql-hooks-ssr`, see https://github.com/nearform/graphql-hooks/blob/master/packages/graphql-hooks-ssr/index.js#L13. Perhaps a comment here to say that `graphql-hooks-ssr` set the flag will make it clearer.

File: packages/graphql-hooks/index.d.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class GraphQLClient {
     options: UseClientRequestOptions
   ): CacheKeyObject
   getFetchOptions(operation: Operation, fetchOptionsOverrides?: object): object
-  request(operation: Operation, options: object): Promise<Result>
+  request(operation: Operation, options?: object): Promise<Result>
```
Comment (bmullan91) [inline packages/graphql-hooks/index.d.ts]: 👍

File: packages/graphql-hooks-ssr/README.md (hunk 1 above)
Comment (helloguille) [reply packages/graphql-hooks-ssr/README.md]: Ok, will fix pull request

**Comment by helloguille** [comment]
Sorry, first time margin a pull request into an open source project.

Is it better I add a commit reversing the ssrMode change or should I start a new pull request with a clean commit history?

**Comment by jackdclark** [comment]
@helloguille another commit is fine, we always squash the commits when merging to master anyway!

**Comment by helloguille** [comment]
Hi:
Fixed to comply with changes requested.

**Comment by bmullan91** [comment]
@all-contributors please add @helloguille for code

**Comment by helloguille** [comment]
@bmullan91 Thanks for merging! I also authored: #406

**Pull request merged**